### DISCUSSION TITLE
Crowdfunding for certificates

### DISCUSSION THREAD

**Discussion by MuhammedKalkan** [MuhammedKalkan/OpenLens#19] [Q&A] [answered]
Hello,

OpenLens builds are working for the most users, however certain conditions causes some setbacks like

- Mac and Windows users can not install if they dont have complete control over the pc
- Auto updaters can not install on Mac
- M1 builds prevents application to run if not signed 

Looking at the issues, most of them are caused by unsigned applications.

In order to overcome this issue, we need developer account from Apple Developer Program and certificates for Windows builds.

We need ideas and see if community wants to support this as suggested here https://github.com/MuhammedKalkan/OpenLens/issues/18#issuecomment-1201057796. Worst case scenario, things will be the way they are now.

1- Are you willing to support these costs by contributing  with the amount you can give ?

2- We will buy a developer account for an individual from apple and cost is 99$ per year.

3- We need to buy certificate for Windows, seems cost is varying. Need options here. Please contribute

4- How would you recommend to collect fees for these ? Patreon ... etc. Please contribute.

**Comment by nistei** [comment]
I would suggest https://opencollective.com/ to make accounting transparent. I've used it for another OSS project before and we gained lots of traction. It can also be linked in GitHub as the default sponsor button.

I think that many companies would be willing to sponsor such a project since it is quite a bit cheaper than the new Lens professional license for a considerable quantity of users.

**Comment by MuhammedKalkan** [comment]
@nistei That seems to be a suitable proposal.  It would be awesome to have you on board to manage opencollective account, so that we can learn from your experience about this platform. Please reach out to me from kalkanikbal@gmail.com for me to be able to add you to as an admin or accountant.

**Comment by MuhammedKalkan** [comment, marked as answer]
Opencollective denied our request for the following reasons

> We don't tend to host projects that are a fork of an existing/currently active project. An exception to this would be if your fork is taking the project in a new direction or is being built to support a feature or platform the original project is no longer supporting.
> 
> My concerns with https://github.com/MuhammedKalkan/OpenLens and https://github.com/lensapp/lens are that the names are the same and there is not enough of an identifiable functional difference on the /MuhammedKalkan/OpenLens called out on the repository to distinguish it from the /lensapp/lens project. 

Therefore, we will move on with the patreon sponsorship. If community backs it up we will go on from there. IF not, binaries are working for most of the users anyways. Thanks for the contributions

**Comment by absmith82** [comment]
Have you looked into [sigstore.dev](https://www.sigstore.dev/).  It's a linux foundation project for free certificates for opensource to solve problems like this.

**Reply by jkroepke** [reply]
Sadly, not trusted by windows systems.

**Comment by MuhammedKalkan** [comment]
I wanted to give some feedback about cert progress. We are already signing apple binaries with an account which belongs to a generous person. About windows cert, i could not obtain one because of some paperwork and waiting for a refund. In  the mean time, if any of the collaborators whishes to buy a signing cert, i can wire the collected amount to that person. Moreover, since we dont need any more donations, i have disabled sponsorship on repo. 
@jkroepke @janhoy

**Reply by jkroepke** [reply]
I can look into this. Which certificate provider should be used here?

**Reply by MuhammedKalkan** [reply]
The one that i have linked in the issue is suitable code signing certificate. Or you can purchase another if you have something in your mind. Just email me and we will talk about transaction

**Reply by jkroepke** [reply]
I contact you

**Comment by MuhammedKalkan** [comment]
About wiring money for certificate. Transaction will happen on Payoneer or Wise. So whomever can lend a hand about  cert, must consider this and have an account on either of these companies.

**Reply by ebbypeter** [reply]
Is the certificate issue resolved or are we still in the process of procuring the cert? If it is not done yet, I can do it.

**Reply by MuhammedKalkan** [reply]
Please reach out via email

**Comment by MuhammedKalkan** [comment]
Alright we now signed the windows binary aswell thanks to @ebbypeter . 20$ remains give or take. Will use it when we need it.

**Reply by PylotLight** [reply]
I'm still having issues with that one on my work laptop.
```
Windows protected your PC
Microsoft Defender SmartScreen prevented an unrecognized app from starting. Running this app might put your PC at risk.
App: 
OpenLens-6.2.1.exe 
Publisher:  
Ebby Peter 
```

**Reply by MuhammedKalkan** [reply]
Applications that are signed with a standard code signing certificates need to have a positive reputation in order to pass the Smart Screen filter. Microsoft establishes the reputation of an executable based upon the number of installations world wide of the same application. Since the signed app is just released (and therefore the reputation hasn't been established as yet), the Smart Screen will continue to flag the application.
